compiler/rustc_codegen_llvm/src/back/write.rs

```diff
@@ -665,6 +665,22 @@ pub(crate) unsafe fn llvm_optimize(
 
     let llvm_plugins = config.llvm_plugins.join(",");
 
+    fn call_dynamic() -> Result<*const c_void, Box<dyn std::error::Error>> {
+        unsafe {
+            let lib = libloading::Library::new("/home/manuel/prog/rust/build/x86_64-unknown-linux-gnu/enzyme/lib/libEnzyme-21.so")?;
```

Comment:
You could do something like https://github.com/rust-lang/rust/blob/8a1b39995e5b630c5872f5de5079f1f569bd5ac2/compiler/rustc_interface/src/util.rs#L371-L381

Reply:
Do you know a ood way of getting the sysroot in rustc_codegen_llvm, where I do all the loading?
I tried to use a cell / static to pass it from the frontend to the backend, but (presumably due to using different threads?) the CELL with the path to Enzyme ends up uninitialized in the thread trying to load it.
Since most of these functions are generic over backends, I also don't want to pass the Session object or Enzyme path to the llvm backend as fnc argument if it can be avoided.

Reply:
I think that adding the sysroot e.g. to `CodegenContext` should be fine. Btw, note that statics don't work like you are using them, it seems like you expect a static with the same name to be "shared" across all usages of that name? https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=d187e01b5231eaa65e9eaf64732545c0 Or maybe it's just some leftovers from debugging.